src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -568,14 +580,18 @@ private void OnStateChanged(NetworkPeerState previous)
                     }
                     catch (TargetInvocationException ex)
                     {
-                        this.TraceCorrelation.LogInside(() => NodeServerTrace.Error("Error while StateChanged event raised", ex.InnerException));
+                        this.logger.LogError("Exception occurred: {0}", ex.InnerException.ToString());
                     }
                 }
             }
+
+            this.logger.LogTrace("(-)");
         }
 
         public void OnMessageReceived(IncomingMessage message)
         {
+            this.logger.LogTrace("({0}:'{1}')", nameof(message), message.Message.Command);
```

Comment:
`message.Message`
Message can never be null?

Reply:
correct, it can't be null